### PR TITLE
fix: fix hashSet deserialize error

### LINKED ISSUE #4544: HashSet  type Deserialize error
https://github.com/elsa-workflows/elsa-core/blob/8c76360f5d6ff9246dbd901ce475ba60b3dba5b7/src/modules/Elsa.Workflows.Core/Serialization/Converters/PolymorphicObjectConverter.cs#L120

Unable to cast object of type 'System.Collections.Generic.HashSet`1[System.String]' to type 'System.Collections.IList'  @sfmskywalker

### PR DESCRIPTION
**Pull request by Nokecy**
close #4544

### CODE REVIEW

File: src/modules/Elsa.Workflows.Core/Serialization/Converters/PolymorphicObjectConverter.cs [hunk 1]
```diff
@@ -130,9 +131,13 @@ public override object Read(ref Utf8JsonReader reader, Type typeToConvert, JsonS
             {
                 array.SetValue(deserializedElement, index++);
             }
-            else
+            else if (collection is HashSet<object> hashSet)
```
Comment (sfmskywalker) [inline src/modules/Elsa.Workflows.Core/Serialization/Converters/PolymorphicObjectConverter.cs:134]: Can we do `ISet` in order to capture more cases?

`if (collection is ISet hashSet)`

Please try it, I'm not sure it will compile or work at runtime.

**Review by sfmskywalker — APPROVED** [review]

File: src/modules/Elsa.Workflows.Core/Serialization/Converters/PolymorphicObjectConverter.cs (hunk 1 above)
Comment (Nokecy) [reply src/modules/Elsa.Workflows.Core/Serialization/Converters/PolymorphicObjectConverter.cs:134]: not found ISet , found ISet<T>

Comment (sfmskywalker) [reply src/modules/Elsa.Workflows.Core/Serialization/Converters/PolymorphicObjectConverter.cs:134]: Sorry, that sentence doesn’t make sense to me. Can you rephrase please?

**Review by sfmskywalker — APPROVED** [review]

File: src/modules/Elsa.Workflows.Core/Serialization/Converters/PolymorphicObjectConverter.cs (hunk 1 above)
Comment (sfmskywalker) [reply src/modules/Elsa.Workflows.Core/Serialization/Converters/PolymorphicObjectConverter.cs]: I understand now, GitHub removed the greater than and less than symbols when you types `Set` and `Set<>`.
